### PR TITLE
Update Rubocop config to match CC styleguide

### PR DESCRIPTION
**Pull request by dblandin**
A small update to bring our rubocop config up-to-date with our styleguide.
Following the refactor in #16, few changes in the codebase were needed.

@codeclimate/review :mag_right:

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -1,43 +1,113 @@
...
+  Exclude:
+  - spec/**/*
+
+Style/ClassAndModuleChildren:
+  Exclude:
+  - spec/**/*
+
+Style/BlockDelimiters:
+  Exclude:
+  - spec/**/*
```
Comment (ABaldwinHunter) [inline .rubocop.yml]: We probably don't need these since we have the top level .codeclimate.yml spec excludes right?

Comment (dblandin) [inline .rubocop.yml]: No you're right. Just moved it to the duplication engine config.

File: .codeclimate.yml [hunk 2]
```diff
@@ -1,4 +1,15 @@
 engines:
+  bundler-audit:
+    enabled: true
+  duplication:
+    enabled: true
+    config:
+      languages:
+        - ruby
```
Comment (ABaldwinHunter) [inline .codeclimate.yml]: :lipstick: un-indent.

File: .codeclimate.yml [hunk 3]
```diff
@@ -1,4 +1,15 @@
...
+    exclude_paths:
+      - spec/
```
Comment (ABaldwinHunter) [inline .codeclimate.yml]: :lipstick: un-indent.

File: .rubocop.yml (hunk 1 above)
Comment (ABaldwinHunter) [inline .rubocop.yml]: ah okay. Did you move it to take advantage of allowing fixme to analyze spec directory?

Comment (dblandin) [inline .rubocop.yml]: Not fixme specifically. I think there's often an acceptable level of duplication in spec files but thought it best to allow other engines, like rubocop and fixme, to scan that directory.

**Comment by ABaldwinHunter** [comment]
Couple of nits, but LGTM!

**Pull request merged**